# This patch file is from Qualcomm Technologies, Inc. and is provided under the following license:
#
# Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
#
# SPDX-License-Identifier: BSD-3-Clause-Clear

From 05a4dcfd7c2e0cc4862e770669d5cd33692655a7 Mon Sep 17 00:00:00 2001
From: Ankurkumar Satasiya <asatasiy@qti.qualcomm.com>
Date: Thu, 25 Sep 2025 22:23:31 +0200
Subject: [PATCH] QcomModulePkg/Library/BootLib: Disable dtbo overlay.

PVM is not using dtbo partition but if dtbo is flashed, bootloader will
try to overlay the device tree with the content of dtbo image and it
will lead to undesired behavior of the guest(most likely the result
will be crash of the guest). So, explicitly disable the dtbo need from
bootloader.

Change-Id: I85480bd8492940677b81009c251beaac61643edc
Signed-off-by: Ankurkumar Satasiya <asatasiy@qti.qualcomm.com>
---
 QcomModulePkg/Library/BootLib/LocateDeviceTree.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/QcomModulePkg/Library/BootLib/LocateDeviceTree.c b/QcomModulePkg/Library/BootLib/LocateDeviceTree.c
index c6dab11c0..8742b9842 100644
--- a/QcomModulePkg/Library/BootLib/LocateDeviceTree.c
+++ b/QcomModulePkg/Library/BootLib/LocateDeviceTree.c
@@ -40,7 +40,7 @@ platform_dt_absolute_match (struct dt_entry *cur_dt_entry,
 STATIC struct dt_entry *
 platform_dt_match_best (struct dt_entry_node *dt_list);
 
-STATIC BOOLEAN DtboNeed = TRUE;
+STATIC BOOLEAN DtboNeed = FALSE;
 
 STATIC INT32 DtboIdx = INVALID_PTN;
 INT32 GetDtboIdx (VOID)
-- 
2.49.0

